app/models/miq_alert.rb

```diff
@@ -193,21 +193,31 @@ def evaluate(target, inputs = {})
     # If we are alerting, invoke the alert actions, then add a status so we can limit how often to alert
     # Otherwise, destroy this alert's statuses for our target
     invoke_actions(target, inputs) if result
-    add_status_post_evaluate(target, result, inputs[:description])
-
+    add_status_post_evaluate(target, result, inputs[:ems_event])
     result
   end
 
-  def add_status_post_evaluate(target, result, status_description)
+  def add_status_post_evaluate(target, result, event)
+    status_description, severity, url = parse_meta(event)
     status = miq_alert_statuses.find_or_initialize_by(:resource => target)
     status.result = result
     status.ems_id = target.try(:ems_id)
     status.description = status_description || description
+    status.severity = severity unless severity.blank?
+    status.url = url unless url.blank?
     status.evaluated_on = Time.now.utc
     status.save
     miq_alert_statuses << status
   end
 
+  def parse_meta(event)
```

Comment:
I'm not sure I understand. The new behavior here is:
- before change: the description of the miq_alert_status always comes from the miq_alert that created it
- after change: The EmsEvent creating the miq_alert_status can have metadata overriding the description. In our use case we take our description from the hawkular alerts (originally defined in the trigger)